docs/docs/concepts/04-aica-applications.md

```diff
@@ -0,0 +1,68 @@
+---
+sidebar_position: 3
+---
+
+# AICA applications
+
+The ROS 2 concepts of managed lifecycle nodes and dynamic composition (and by extension, lifecycle components) enable
+more stateful and modular building blocks for complex applications. A lifecycle component can be flexibly instantiated
+and destroyed at run-time, and its lifecycle states and transitions can be managed according to key application states
+or events.
+
+Similarly, the `ros2_control` architecture allows controllers to be dynamically loaded and activated or deactivated for
+a given hardware interface, enabling reactive and versatile robot behaviors at runtime.
+
+The catch is that components and controllers need to be _managed_; it is the job of some supervisor process
+or processes to monitor states and trigger the corresponding events to load, activate, deactivate or unload components,
+controllers or hardware interfaces.
+
+All the components and controllers in an application also need to be communicating on matching interfaces; in ROS 2
+terminology, they must have matching topic names and message types.
+
+Building an advanced robotic application then results in two key types of challenge:
+
+- designing and implementing the task-specific conditions and events that manage component and controller states
+- arranging compatible components and controllers in a network graph of communication channels and messages
+
+If application-specific state logic or graph structure is hard-coded into one or more components, those components cease
+to be modular or re-usable for other applications. On the other hand, programming and compiling the application logic
+and graph into a dedicated supervisor program generally undermines the modern concepts of abstraction and dynamic
+reconfiguration that these modern ROS 2 tools provide.
+
+To address this challenge, the AICA framework includes at its core a generalized execution process known as the dynamic
+state engine.
+
+## The dynamic state engine
+
+The dynamic state engine is responsible for managing the states of all the elements of an AICA application.
+
+An AICA application is an interconnected graph of **components**, **controllers** and **hardware interfaces**.
+Components process data in a periodic step function and transfer data as **signals** to other components and
+controllers. Hardware interfaces are drivers that connect controllers to robots.
+
+The dynamic state engine manages the application state using **events**, which are used to reconfigure parameters, load
+or unload components, manage component lifecycle states, trigger specific service actions and more.
+
+## Declarative application descriptions
```

Comment:
nitpick: Even though it's the correct word, I find 'declarative' a bit heavy in this context. Is an adjective even needed?

Reply:
To me, declarative is very important in this context. It refers to the programming paradigm, and the implementation / program flow is abstract.

I also wouldn't want to have just `## Application descriptions` or `## Application syntax` on its own as a section header.

I prefer the term "application descriptions" which can include both YAML and graph forms. Application syntax refers more specifically to the YAML. But, both the visual and scripted descriptions are forms of declarative programming.